File: src/app/Class/Genre.ts
import { Country } from './Country';

export class Genre {
  id: number;
  origin: Country;
  genreName: string;
  creationDate: string;
  aboutGenre: string;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.origin = (obj && obj.origin) || null;
    this.genreName = (obj && obj.genreName) || '';
    this.creationDate = (obj && obj.creationDate) || '';
    this.aboutGenre = (obj && obj.aboutGenre) || '';
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getOrigin(): Country {
    return this.origin;
  }

  getGenreName(): string {
    return this.genreName;
  }

  getCreationDate(): string {
    return this.creationDate;
  }

  getAboutGenre(): string {
    return this.aboutGenre;
  }

  // Setters

  setOrigin(origin: Country) {
    this.origin = origin;
  }

  setGenreName(genreName: string) {
    this.genreName = genreName;
  }

  setCreationDate(creationDate: string) {
    this.creationDate = creationDate;
  }

  setAboutGenre(aboutGenre: string) {
    this.aboutGenre = aboutGenre;
  }
}

File: Backend/src/main/java/musiccube/entities/Genre.java
package musiccube.entities;

import com.fasterxml.jackson.annotation.JsonInclude;

import javax.persistence.*;

@Entity
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Genre {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    private Country origin;

    @Column(unique = true)
    private String genreName;

    @Column
    private String creationDate;

    @Column
    private String aboutGenre;

    public Genre() {}

    public Genre(Country origin, String genreName, String creationDate, String aboutGenre) {
        this.origin = origin;
        this.genreName = genreName;
        this.creationDate = creationDate;
        this.aboutGenre = aboutGenre;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Country getOrigin() {
        return origin;
    }

    public void setOrigin(Country origin) {
        this.origin = origin;
    }

    public String getGenreName() {
        return genreName;
    }

    public void setGenreName(String genreName) {
        this.genreName = genreName;
    }

    public String getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(String creationDate) {
        this.creationDate = creationDate;
    }

    public String getAboutGenre() {
        return aboutGenre;
    }

    public void setAboutGenre(String aboutGenre) {
        this.aboutGenre = aboutGenre;
    }
}
